rclcpp/test/rclcpp/CMakeLists.txt

```diff
@@ -262,6 +267,11 @@ ament_add_gtest(test_node_interfaces__node_waitables
 if(TARGET test_node_interfaces__node_waitables)
   target_link_libraries(test_node_interfaces__node_waitables ${PROJECT_NAME} mimick)
 endif()
+add_library(test_node_interfaces__test_template_utils  # Compile time test
+  node_interfaces/detail/test_template_utils.cpp)
+if(TARGET test_node_interfaces__test_template_utils)
+  target_link_libraries(test_node_interfaces__test_template_utils ${PROJECT_NAME})
+endif()
```

Comment:
I think we need to check some other things that `ament_add_gtest()` does for you. For example, making sure it's not part of the "all" target and instead part of the "test" target, etc.

Maybe all we need is https://cmake.org/cmake/help/latest/prop_tgt/EXCLUDE_FROM_ALL.html and then add it to the `test` make target. I'm not sure what else might be needed. Might want to ask someone else on the ros team to review it.

Reply:
Resolved with a no-op gtest